### PR TITLE
Workaround MSAL.NET issue with MSA-PT account silent auth

### PR DESCRIPTION
**Pull request by mjcheetham**
When we have a Microsoft Account (MSA) in the cache and attempt to do a silent authentication, if we're an MSA-PT app we need to specify the special MSA transfer tenant ID to make sure we get the a token silently, correctly.

See the [issue](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3077) in the MSAL repo for more information.

Fixes: #1297

### CODE REVIEW

File: src/shared/Core/Authentication/MicrosoftAuthentication.cs [hunk 1]
```diff
@@ -295,9 +297,27 @@ private async Task<AuthenticationResult> GetAccessTokenSilentlyAsync(IPublicClie
...
+                    if (account is null)
+                    {
+                        Context.Trace.WriteLine($"No cached account found for user '{userName}'...");
+                        return null;
+                    }
+
+                    var atsBuilder = app.AcquireTokenSilent(scopes, account);
+
+                    // Is we are operating with an MSA passthrough app we need to ensure that we target the
+                    // special MSA 'transfer' tenant explicitly. This is a workaround for MSAL issue:
```
Comment (ldennington) [inline src/shared/Core/Authentication/MicrosoftAuthentication.cs:312]: Super nit: I'm wondering whether there's any public documentation available about the transfer tenant that we can link here for those who would like to learn more.

**Review by ldennington — APPROVED** [review]
Nice! ✨

File: src/shared/Core/Authentication/MicrosoftAuthentication.cs (hunk 1 above)
Comment (mjcheetham) [reply src/shared/Core/Authentication/MicrosoftAuthentication.cs:312]: Unfortunately MSA passthrough is an internal (and legacy) feature of the Microsoft Identity platform, and the only docs are for Microsoft employees only: https://review.learn.microsoft.com/en-us/identity/microsoft-identity-platform/stack-choices?branch=main#msa-passthrough

**Comment by mohamadfaqani** [comment]
***@***.***

Hello, I don't know why, but I don't receive emails in my Microsoft
account, no one knows how to fix it

gmail.android

**Review by ldennington — APPROVED** [review]

**Pull request merged**